File: 250805/키를 기준으로 정렬/sort-by-height.py
class Students :
    def __init__(self, name, height, weight) :
        self.name = name;
        self.height = height;
        self.weight = weight;
    


n = int(input());

students = []

for _ in range(n) :
    name, height, weight = tuple(input().split());
    students.append(Students(name, int(height),int(weight)));

students.sort(key = lambda x: x.height)


for elem in students :
    print(elem.name, elem.height, elem.weight)

File: 250807/키를 기준으로 정렬/sort-by-height.java
import java.util.*;

public class Main {
    static class Students implements Comparable<Students>{

        String name;
        int height;
        int weight;

        public Students(String name, int height, int weight){
            this.name = name;
            this.height = height;
            this.weight = weight;
        }

        @Override
        public int compareTo(Students students) {
            return this.height - students.height;            
        }
    }

    public static void main(String[] args) {
        // Please write your code here.

        Scanner sc = new Scanner(System.in);

       int N = sc.nextInt();
       
       String name = "";
       int height = 0;
       int weight = 0;


        Students students[] = new Students[N];

        for (int i = 0; i < N; i++) {
            
        name = sc.next();
        height = sc.nextInt();
        weight = sc.nextInt();

        students[i] = new Students(name, height, weight);

        }

        Arrays.sort(students);

        for (int i = 0; i < N; i ++){

            System.out.println(students[i].name + " " + students[i].height + " " + students[i].weight);
        }


    }
}